scripts/bazel_diff.py

```diff
@@ -0,0 +1,205 @@
+
+Computes the rules that differ between a provided baseline commit and the
+current git commit.
+"""
+
+__copyright__ = """
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""
+
+import subprocess
+import argparse
+import tempfile
+import os
+import sys
+from pathlib import Path
+from typing import List
+
+import scripts_utils
+
+
+def log(s: str) -> None:
+    print(s, file=sys.stderr)
+
+
+def quiet_run(args: List[str]) -> None:
+    try:
+        subprocess.run(
+            args,
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.PIPE,
+            check=True,
+            encoding="utf-8",
+        )
+    except subprocess.CalledProcessError as err:
+        log(err.stderr)
+        raise
+
+
+def quiet_run_output(args: List[str]) -> str:
+    try:
+        p = subprocess.run(
+            args,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            check=True,
+            encoding="utf-8",
+        )
+        return p.stdout
+    except subprocess.CalledProcessError as err:
+        log(err.stderr)
+        raise
+
+
+def make_bazel_diff_script(bazel: Path, tmpdir: Path) -> Path:
+    bazel_diff_path = tmpdir / "bazel_diff"
+    quiet_run(
+        [
+            str(bazel),
+            "run",
+            f"--script_path={bazel_diff_path}",
+            "//bazel/diff:bazel-diff",
+        ]
+    )
+    return bazel_diff_path
+
+
+def compute_hashes(
+    bazel: Path, bazel_diff: Path, tmpdir: Path, prefix: str
+) -> Path:
+    hashes_path = tmpdir / f"{prefix}_hashes"
+    quiet_run(
+        [
+            str(bazel_diff),
+            "generate-hashes",
+            f"-b={bazel}",
+            f"-w={os.getcwd()}",
+            str(hashes_path),
+        ]
+    )
+    return hashes_path
+
+
+def impacted_targets(
+    bazel_diff: Path, baseline_hashes: Path, current_hashes: Path
+) -> str:
+    return quiet_run_output(
+        [
+            str(bazel_diff),
+            "get-impacted-targets",
+            f"-sh={baseline_hashes}",
+            f"-fh={current_hashes}",
+        ]
+    )
+
+
+def filter_targets(bazel: Path, targets: str) -> str:
+    with tempfile.NamedTemporaryFile(mode="w+") as tmp:
+        tmp.write(
+            f"let t = set({targets}) in "
+            "kind(rule, $t) except attr(tags, manual, $t)\n"
+        )
+        tmp.seek(0)
+        tmp_head = "".join(line for (line, _) in zip(tmp, range(10)))
+        if tmp.read(1) != "":
+            tmp_head += "...\n"
+        log(f"Bazel query file's first 10 lines:\n{tmp_head}---")
+        return quiet_run_output(
+            [
+                str(bazel),
+                "query",
+                f"--query_file={tmp.name}",
+            ]
+        )
+
+
+def git_is_dirty() -> bool:
+    output = subprocess.check_output(
```

Comment:
You're mixing check_output with run (line 131). You'd pointed me to documentation saying run is preferred, should you use throughout?